src/snowflake/connector/connection.py

```diff
@@ -329,183 +331,184 @@ def _ocsp_mode(self) -> OCSPMode:
             return OCSPMode.FAIL_CLOSED
 
     @property
-    def session_id(self):
+    def session_id(self) -> int:
         return self._session_id
 
     @property
-    def user(self):
+    def user(self) -> str:
         return self._user
 
     @property
-    def host(self):
+    def host(self) -> str:
         return self._host
 
     @property
-    def port(self):
+    def port(self) -> int | str:  # TODO: shouldn't be a string
         return self._port
 
     @property
-    def region(self):
+    def region(self) -> str | None:
         warnings.warn(
             "Region has been deprecated and will be removed in the near future",
             PendingDeprecationWarning,
         )
         return self._region
 
     @property
-    def proxy_host(self):
+    def proxy_host(self) -> str | None:
         return self._proxy_host
 
     @property
-    def proxy_port(self):
+    def proxy_port(self) -> str | None:
         return self._proxy_port
 
     @property
-    def proxy_user(self):
+    def proxy_user(self) -> str | None:
         return self._proxy_user
 
     @property
-    def proxy_password(self):
+    def proxy_password(self) -> str | None:
         return self._proxy_password
 
     @property
-    def account(self):
+    def account(self) -> str:
         return self._account
 
     @property
-    def database(self):
+    def database(self) -> str | None:
         return self._database
 
     @property
-    def schema(self):
+    def schema(self) -> str | None:
         return self._schema
 
     @property
-    def warehouse(self):
+    def warehouse(self) -> str | None:
         return self._warehouse
 
     @property
-    def role(self):
+    def role(self) -> str | None:
         return self._role
 
     @property
-    def login_timeout(self):
+    def login_timeout(self) -> int | None:
         return int(self._login_timeout) if self._login_timeout is not None else None
 
     @property
-    def network_timeout(self):
+    def network_timeout(self) -> int | None:
         return int(self._network_timeout) if self._network_timeout is not None else None
 
     @property
-    def client_session_keep_alive(self):
+    def client_session_keep_alive(self) -> bool | None:
         return self._client_session_keep_alive
 
     @client_session_keep_alive.setter
-    def client_session_keep_alive(self, value):
+    def client_session_keep_alive(self, value) -> None:
```

Comment:
I saw you gave type hint to the `value` of the `errorhandler` setter.
are we missing the type hint for `value` of other setters?